### ISSUE TITLE
GitHub workflows bundle deploy fails with missing terraform binary and terraform.tfstate

### ISSUE THREAD

**Issue by sebwiendl** [databricks/setup-cli#119]
Hello,

I'm currently in the process of setting up CI/CD for our first Databricks DLT pipeline. My workflow looks like this:
```
jobs:
  deploy-asset-bundle:
    name: Deploy Asset Bundle ${{ inputs.BUNDLE_ROOT }} ${{ inputs.STAGE }}
    runs-on: ubuntu-latest
    env:
      BUNDLE_ROOT: ${{ inputs.BUNDLE_ROOT }}
      DATABRICKS_HOST: ${{ secrets.*** }}
      DATABRICKS_CLIENT_ID: ${{ secrets.*** }}
      DATABRICKS_CLIENT_SECRET: ${{ secrets.*** }}
    steps:
      - uses: actions/checkout@v4
      - uses: databricks/setup-cli@main
      - run: databricks bundle validate -t ${{ inputs.STAGE }}
      - run: databricks bundle deploy -t ${{ inputs.STAGE }}
```
When running the workflow, the CLI is setup successfully, validate runs successfully, and the files are uploaded successfully to our workspace before failing in the "Deploying resources..." step:
```
Run databricks bundle deploy -t dev
  databricks bundle deploy -t dev
  shell: /usr/bin/bash -e {0}
  env:
    BUNDLE_ROOT: ***
    DATABRICKS_HOST: ***
    DATABRICKS_CLIENT_ID: ***
    DATABRICKS_CLIENT_SECRET: ***
Uploading bundle files to /Users/***/.bundle/***/dev/files...
Deploying resources...
Error: terraform init: fork/exec ***/.databricks/bundle/dev/bin/terraform: no such file or directory

Error: open ***/.databricks/bundle/dev/terraform/terraform.tfstate: no such file or directory
```
The terraform binary and terraform.tfstate file are available locally (initialized with `databricks bundle init`), but ignored by default via `.gitignore` and therefore not part of the repo.

How can I deploy an asset bundle via GitHub workflow? My solution closely follows the [docs](https://docs.databricks.com/en/dev-tools/bundles/ci-cd.html).

**Comment by pietern** [comment]
Thanks for reporting.

The binaries don't need to be part of the repository. They are downloaded automatically if they're not present. Could you run the commands (validate and deploy) with the `--debug` flag and see if anything stands out?

Are you using a GitHub-hosted or self-hosted actions runner? If the latter, are there firewalls in place?

**Comment by sebwiendl** [comment]
Hi,

thanks for the fast feedback. In the meantime, I figured it out: I was using a relative BUNDLE_ROOT, which failed. I changed it (for testing) to a hardcoded BUNDLE_ROOT which worked as expected. Now i only have to dynamically construct my BUNDLE_ROOT to be absolute.

Did a `tree -a` and indeed the `terraform` binary is present on the runner.

The relative BUNDLE_ROOT worked fine locally. There seems to be some difference in behavior from local to GitHub workflows.
